rcljava/src/main/java/org/ros2/rcljava/events/EventHandlerImpl.java

```diff
@@ -121,8 +101,12 @@ public final long getHandle() {
    * {@inheritDoc}
    */
   public final void dispose() {
-    nativeDispose(this.handle);
-    this.handle = 0;
+    synchronized(this.handleMutex) {
```

Comment:
Instead of creating a separate object for the mutex, why not use the `EventHandlerImpl` instance itself? I.e. we can declare this method and `executeCallback` as `synchronized`.

Reply:
Fixed in https://github.com/osrf/ros2_java/pull/9/commits/fb342cca664e8df957bed7c0377df565c8bac38c.